wordpress-shortcodes/src/main/java/org/wordpress/aztec/plugins/shortcodes/extensions/CaptionExtensions.kt

```diff
@@ -44,17 +45,20 @@ fun AztecText.removeImageCaption(attributePredicate: AztecText.AttributePredicat
 
 fun AztecText.hasImageCaption(attributePredicate: AztecText.AttributePredicate): Boolean {
     this.text.getSpans(0, this.text.length, AztecImageSpan::class.java)
-        .firstOrNull {
-            val wrapper = SpanWrapper<AztecImageSpan>(text, it)
-            return text.getSpans(wrapper.start, wrapper.end, CaptionShortcodeSpan::class.java).isNotEmpty()
+            .firstOrNull {
+                attributePredicate.matches(it.attributes)
```

Comment:
Hey @0nko , just to be sure, does the filtering contribute to the fix of #775 ? Asking just to understand if there's some more code paths I should review and test. Thanks!

Reply:
No. I just noticed that the function is not using the parameter and it wasn't working as expected. It's not even being used anywhere at the moment, just in the unit tests. I know PRs should be specific but since I was already working on the file I wanted to make a quick fix. We can remove it if you prefer.

Reply:
Ah, cool. No, we don't need to remove it, it's fine. Just wanted to make sure I'm not missing something. Looks good to me!